kafka/consumer/Dockerfile

```diff
@@ -3,14 +3,15 @@
 # Licensed under the MIT License. See License.txt in the project root for license information.
 # --------------------------------------------------------------------------------------------
 
-FROM golang:1.21 as builder
-WORKDIR /app
-COPY go.* ./
-RUN go mod download
-COPY main.go ./
-RUN CGO_ENABLED=0 GOOS=linux go build -mod=readonly  -v -o consume
+FROM golang:1.22 as builder
+ENV PROJECT_DIR=/go/src/github.com/microsoft
+WORKDIR ${PROJECT_DIR}/confidential-container-demos
+
+COPY /kafka/consumer ./kafka/consumer
+COPY /kafka/util ./kafka/util
+RUN cd kafka/consumer && go mod download && CGO_ENABLED=0 GOOS=linux go build -mod=readonly  -v -o consume
```

Comment:
We shouldn't need the download step here.  The vendor directory already exists in our source path and should be copied into the container instead of downloaded.